processing/src/main/java/io/druid/query/extraction/namespace/ExtractionNamespaceFunctionFactory.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction.namespace;
+
+import com.google.common.base.Function;
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+/**
+ *
+ */
+public interface ExtractionNamespaceFunctionFactory<T extends ExtractionNamespace>
```

Comment:
I'm not convinced of the way this interface is structured.  Who is intended to use it and for what purpose?

Fwiw, my mental image for how this stuff should be handled is something like how the `ZkCoordinator` handles things.  That is, there should be a class (let's call it `NamespaceManager`) that is given objects (let's call them `NamespaceSpec`s) which can load up a namespace on their own.  The `NamespaceManager` then maintains a map of namespace name to the thing that was loaded up by the `NamespaceSpec`.  It has a method `getNamespace(String)` that can be used to access a namespace.